routers/api/v1/activitypub/reqsignature.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package activitypub
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"code.gitea.io/gitea/modules/activitypub"
+	gitea_context "code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/httplib"
+	"code.gitea.io/gitea/modules/setting"
+
+	ap "github.com/go-ap/activitypub"
+	"github.com/go-fed/httpsig"
+)
+
+func getPublicKeyFromResponse(ctx context.Context, b []byte, keyID *url.URL) (p crypto.PublicKey, err error) {
+	person := ap.PersonNew(ap.IRI(keyID.String()))
+	err = person.UnmarshalJSON(b)
+	if err != nil {
+		err = fmt.Errorf("ActivityStreams type cannot be converted to one known to have publicKey property: %T", b)
```

Comment:
Always try to include the error.
```suggestion
		err = fmt.Errorf("ActivityStreams type cannot be converted to one known to have publicKey property: %v", err)
```

`%T` in this case would always return `[]uint8` because b isn't a `interface{}` or alike.